src/rectangle.rs

```diff
@@ -32,3 +89,46 @@ impl PartialEq for Rect {
         self.ll == other.ll && self.ur == other.ur
     }
 }
+
+/// The paint mode of a [`Rect`].
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum Mode {
```

Comment:
Do you want me to leave `PolygonMode` and `WindingOrder` in the current module? As I pointed out before, it would make sense to have a unique version, but on the other hand it would make sense to move to a _common_ module and, in the case of the `PolygonMode`, rename it IMHO.

Reply:
yeah, move it out if it's shared between modules